### ISSUE TITLE
[Java] A false negative about IL_INFINITE_LOOP

### ISSUE THREAD

**Issue by Zustin** [spotbugs/spotbugs#1641]
Hi, when using SpotBugs to improve my code quality, I found a false negative about the rule [IL_INFINITE_LOOP](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#il-an-apparent-infinite-loop-il-infinite-loop). The minimal case is below:

```java
public boolean getCondition() {
    return true;
}
public void methodA() {
    boolean condition = getCondition();
    if(condition) {
        methodA();  // should report a warning here
    } else {
        methodA();  // And report a warning here
    }
}
```
Besides, I found that SpotBugs repo contains a testing file about this rule([Link](https://github.com/spotbugs/spotbugs/blob/51e586bed98393e53559a38c1f9bd15f54514efa/spotbugsTestCases/src/java/sfBugs/Bug3073124.java)). In the end of this file, it provides a similar test case at line 41.

Version: using IntelliJ SpotBugs plugin 1.2.4 with SpotBugs version 4.1.3

**Comment by ThrawnCA** [comment]
I don't see any loop in your code. It would make sense to trigger a detector based on both branches having the same code, but why would it trigger IL_INFINITE_LOOP?

**Comment by Zustin** [comment]
@ThrawnCA  Hi, I am sorry that I gave the wrong rule. It should have been IL_INFINITE_RECURSIVE_LOOP. I have fixed it. Thanks for your reply and consideration.

**Comment by Zustin** [comment]
@KengoTODA Hi Kengo, could you please help check this issue? Thanks for your kind help.
